tests/src/core/testqgsexpression.cpp

```diff
@@ -1663,8 +1663,24 @@ class TestQgsExpression: public QObject
       QTest::newRow( "array_majority('forty two')" ) << QStringLiteral( "array_majority('forty two')" ) << true << QVariant();
       QTest::newRow( "array_majority(42)" ) << QStringLiteral( "array_majority(42)" ) << true << QVariant();
-      QTest::newRow( "array_majority(array())" ) << QStringLiteral( "array_majority(array())" ) << false << QVariant();
-      QTest::newRow( "array_majority(array(1,2,42,42,'a','b'))" ) << QStringLiteral( "array_majority(array(1,2,42,42,'a','b'))" ) << false << QVariant( QVariantList() << 42 );
+      QTest::newRow( "array_majority(array())" ) << QStringLiteral( "array_majority(array())" ) << false << QVariant( QVariantList() );
+      QTest::newRow( "array_majority(array(0,1,42,42,43), 'all')" ) << QStringLiteral( "array_majority(array(0,1,42,42,43), 'all')" ) << false << QVariant( QVariantList() << "42" );
+      QTest::newRow( "array_majority(array(0,1,42,42,43,1)" ) << QStringLiteral( "array_sort(array_majority(array(0,1,42,42,43,1)))" ) << false << QVariant( QVariantList() << 1 << 42 );
+      QTest::newRow( "array_majority(array(0,1,42,42,43,1), 'first')" ) << QStringLiteral( "array_majority(array(0,1,42,42,43), 'first')" ) << false << QVariant( 42 );
+      QTest::newRow( "array_majority(array(0,1,1,2,2,42), 'median')" ) << QStringLiteral( "array_majority(array(0,1,1,2,2,42), 'median')" ) << false << QVariant( 1.5 );
+      QTest::newRow( "array_majority(array(0,1,42,42,43), 'real_majority')" ) << QStringLiteral( "array_majority(array(0,1,42,42,43), 'real_majority')" ) << false << QVariant();
+      QTest::newRow( "array_majority(array(0,1,42,42,43,42), 'real_majority')" ) << QStringLiteral( "array_majority(array(0,1,42,42,43,42), 'real_majority')" ) << false << QVariant();
+      QTest::newRow( "array_majority(array(0,1,42,42,43,42,42), 'real_majority')" ) << QStringLiteral( "array_majority(array(0,1,42,42,43,42,42), 'real_majority')" ) << false << QVariant( 42 );
+      QTest::newRow( "array_minority('forty two')" ) << QStringLiteral( "array_minority('forty two')" ) << true << QVariant();
+      QTest::newRow( "array_minority(42)" ) << QStringLiteral( "array_minority(42)" ) << true << QVariant();
+      QTest::newRow( "array_minority(array())" ) << QStringLiteral( "array_minority(array())" ) << false << QVariant( QVariantList() );
+      QTest::newRow( "array_minority(array(0,42,42), 'all')" ) << QStringLiteral( "array_minority(array(0,42,42), 'all')" ) << false << QVariant( QVariantList() << 0 );
+      QTest::newRow( "array_minority(array(0,1,42,42))" ) << QStringLiteral( "array_sort(array_minority(array(0,1,42,42)))" ) << false << QVariant( QVariantList() << 0 << 1 );
+      QTest::newRow( "array_minority(array(0,42,42), 'first')" ) << QStringLiteral( "array_minority(array(0,42,42), 'first')" ) << false << QVariant( 0 );
+      QTest::newRow( "array_minority(array(0,1,2,3,3), 'median')" ) << QStringLiteral( "array_minority(array(1,2,3,3), 'median')" ) << false << QVariant( 1.5 );
```

Comment:
Since we are doing some math here, should we have test with non numeric values? *ps: I don't know what I'm talking about*

Reply:
I am not sure what you mean? Should I add some more test with non numeric values? This functions are more statistic than math, should work with every type, in a given set if a number or a word as majority it makes no difference, is just about counting the occurrences of each value. For example `array_majority(array(0,1,42,42,43,'a','a'), 'all')` returns [ 42, 'a' ]

Reply:
> This functions are more statistic than math
> is just about counting the occurrences of each value

Isn't the median calculated (I fail to see how 1.5 can be a count) from the "all" candidates? Then what happens if a string is among those candidates? I'm also asking because I think there was a test with string values, now removed.
PS: don't forget the first ps.

Reply:
Yes, please add tests with non-numeric values too!

Reply:
> > This functions are more statistic than math
> > is just about counting the occurrences of each value
> 
> Isn't the median calculated (I fail to see how 1.5 can be a count) from the "all" candidates? Then what happens if a string is among those candidates? I'm also asking because I think there was a test with string values, now removed.
> PS: don't forget the first ps.

You are right @DelazJ , by 'median' the values are passed to the function array_median which ignores non arithmetic values. In case the values are even the median is calculated as the mean of the two middle values.
Probably I should also add this to the documentation.